docs/user_guides/Gufi For Coders/main.tex

```diff
@@ -0,0 +1,226 @@
+\subsection{Flags}
+
+\begin{table} [h]
+\centering
+\begin{tabular}{l|p{6cm}}
+Flag & Functionality \\\hline
+-h & help\\
+\hline
+-H & show assigned input values (debugging)\\
+\hline
+-E \textless SQL ent\textgreater & SQL for entries table \\
+\hline
+-S \textless SQL sum\textgreater (will be read first) & SQL for summary table\\
+\hline
+-T \textless SQL tsum\textgreater & SQL for tree-summary table\\
+\hline
+-a & AND/OR (SQL query combination)\\
+\hline
+-n \textless threads\textgreater & number of threads\\
+\hline
+-j & print the information in terse form\\
+\hline
+-o \textless out\_fname\textgreater & output file (one-per thread, with thread-id suffix) implies -e 1\\
+\hline
+-d \textless delim\textgreater & one char delimiter \\
+\hline
+-O \textless out\_DB\textgreater & output DB, implies -e 1 \\
+\hline
+-I \textless SQL\_init\textgreater & SQL init \\
+\hline
+-F \textless SQL\_fin\textgreater & SQL cleanup \\
+\hline
+-y \textless min-level\textgreater & minimum level to descend to \\
+\hline
+-z \textless max-level\textgreater & maximum level to descend to\\
+\hline
+-J \textless SQL\_interm\textgreater & SQL for intermediate results (no default: recommend using "SELECT * FROM entries") \\
+\hline
+-K \textless create aggregate\textgreater & SQL to create the final aggregation table (if not specified, -I will be used)\\
+\hline
+-G \textless SQL\_aggregate\textgreater & SQL for aggregated results (no default: recommend using "SELECT * FROM entries")\\
+\hline
+-e \textless 0 or 1\textgreater & 0 for aggregate, 1 for print without aggregating (implied by -o and -O)\\
+\hline
+-m & Keep mtime and atime same on the database files \\
+\hline
+-B \textless buffer size\textgreater & size of each thread's output buffer in bytes \\
+\hline
+-w & open the database files in read-write mode instead of read only mode
+\end{tabular}
+\caption{\label{tab:widgets}Flags and Arguments}
+\end{table}
+
+\clearpage
+
+\subsection{Example Calls}
+
+gufi\_query -S "SELECT * FROM summary" ~/directory\_of\_root\_index
+\\
+gufi\_query -E "SELECT * FROM pentries" ~/directory\_of\_root\_index
+
+
+\subsection{Visualizing the Workflow}
+
+
+\begin{figure} [h]
+\centering
+\includegraphics[width=1.0\textwidth]{gufi_query.png}
+\caption{\label{fig:gufi_query}Workflow of gufi\_query}
+\end{figure}
+
+\subsection{Processdir}
+What you will quickly notice when tracing through the above diagram (Figure 3) the major functionality from this code comes from the processdir function. This function is where the -E -S and -T flags are processed. Therefore, this is where all of the SQL database accesses occur. This process is parallelized via a thread pool and is further optimized by a calculated rollupscore to ensure that we only descend into necessary sub-directories.
```

Comment:
> Therefore, this is where all of the SQL database accesses occur.

These queries access the index, but there are other queries that occur when aggregating.